internal/pkg/docker/dockerfile/ignorefile.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockerfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/moby/patternmatcher/ignorefile"
+	"github.com/spf13/afero"
+)
+
+// DockerignoreFile represents a .dockerignore file holding information of excluded files.
+type DockerignoreFile struct {
+	excludes []string
+
+	fs afero.Fs
+}
+
+// NewDockerignoreFile returns a DockerignoreFile read from a given filesystem.
+func NewDockerignoreFile(fs afero.Fs, contextDir string) (*DockerignoreFile, error) {
+	df := &DockerignoreFile{
+		fs:       fs,
+		excludes: []string{},
+	}
+	return df, df.readDockerignore(contextDir)
+}
+
+// ReadDockerignore reads the .dockerignore file in the context directory and
+// returns the list of paths to exclude.
+func (df *DockerignoreFile) readDockerignore(contextDir string) error {
+	f, err := df.fs.Open(filepath.Join(contextDir, ".dockerignore"))
+	switch {
+	case os.IsNotExist(err):
+		return nil
+	case err != nil:
+		return err
+	}
```

Comment:
```suggestion
	if err != nil {
	  if os.IsNotExist(err) {
	    return nil
	  }
		return err
	}
```
Does this make more sense?

Reply:
Personally I don't like checking the error in a nested way

Reply:
It's logically different I think: we have to depend on `os.IsNotExist` being able to deal with the `nil` case in your current impl. But i think overall it's not a big deal.

Reply:
I see the difference now, works for me 👍